frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -305,6 +315,140 @@ pub mod pallet {
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
+        /// Lock up and stake balance of the origin account.
+        ///
+        /// `value` must be more than the `minimum_balance` specified by `T::Currency`
+        /// unless account already has bonded value equal or more than 'minimum_balance'.
+        ///
+        /// The dispatch origin for this call must be _Signed_ by the staker's account.
+        ///
+        /// Effects of staking will be felt at the beginning of the next era.
+        ///
+        /// TODO: Weight!
+        #[pallet::weight(10)] // TODO: fix this later. Probably a new calculation will be required since logic was changed significantly.
+        pub fn bond_and_stake(
+            origin: OriginFor<T>,
+            contract_id: SmartContract<T::AccountId>,
+            #[pallet::compact] value: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let staker = ensure_signed(origin)?;
+            ensure!(
+                Self::is_contract_valid(&contract_id),
+                Error::<T>::ContractIsNotValid
+            );
+            ensure!(
+                RegisteredDapps::<T>::contains_key(&contract_id),
+                Error::<T>::NotOperatedContract
+            );
+
+            // Get the staking ledger or create an entry if it doesn't exist.
+            let mut ledger = if let Some(ledger) = Self::ledger(&staker) {
+                ledger
+            } else {
+                StakingLedger {
+                    stash: staker.clone(),
+                    total: Zero::zero(),
+                    active: Zero::zero(),
+                    unlocking: vec![],
+                    last_reward: Zero::zero(),
+                }
+            };
+
+            // Ensure that staker has enough balance to bond & stake.
+            let free_balance = T::Currency::free_balance(&staker);
+            // Remove already locked funds from the free balance
+            let available_balance = free_balance
+                .checked_sub(&ledger.total)
```

Comment:
Please explain this line

Reply:
You mean in the code or here?

I believe you did the same thing when implementing bond(...) method.
Funds are locked but they are still calculated into free_balance. However, free_balance cannot fall below a threshold which is equal to locked funds. E.g. you have 1000 tokens, you've locked 300, you still have 1000 free_balance but can only spend 700.